python/pyspark/sql/column.py

```diff
@@ -1524,7 +1524,11 @@ def over(self, window: "WindowSpec") -> "Column":
     @dispatch_col_method
     def outer(self) -> "Column":
         """
-        Mark this column reference as an outer reference for subqueries.
+        Mark this as an outer Column if its expression may reference columns in the outer plan.
```

Comment:
```suggestion
        Mark this column as an outer column if its expression refers to columns from an outer query or plan
```

Reply:
do you mean `outer query plan`?

Reply:
Yea, or just `outer query`. This is going to be in the API doc, but we can always refine it later.